lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -188,13 +188,23 @@ void SPIRVRegularizeLLVMBase::lowerMemset(MemSetInst *MSI) {
   return;
 }
 
-void SPIRVRegularizeLLVMBase::buildFunnelShiftLeftFunc(Function *FSHLFunc) {
-  if (!FSHLFunc->empty())
+void SPIRVRegularizeLLVMBase::lowerFunnelShift(IntrinsicInst *FSHIntrinsic) {
+  // Get a separate function - otherwise, we'd have to rework the CFG of the
+  // current one. Then simply replace the intrinsic uses with a call to the new
+  // function.
```

Comment:
Can we also add a comment with expected LLVM IR for the function body?

Reply:
I mean body of `@spirv.llvm_fshl_` and `@spirv.llvm_fshr_`. 
It is easier to read the code if we see what we are generating.